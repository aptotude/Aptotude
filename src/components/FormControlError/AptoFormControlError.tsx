import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoFormControlError.scss';

const COMPONENT_PREFIX = 'AptoFormControlError';

export interface AptoFormControlErrorDisplayProps extends StandardTypes {
  error?: string | undefined | null;
}

export class AptoFormControlError extends React.Component<
  AptoFormControlErrorDisplayProps
> {
  public static defaultProps = {
    error: undefined
  };

  public render() {
    const { error, forwardRef, className, ...rest } = this.props;

    const classes = classNames(COMPONENT_PREFIX, className);

    if (!error) {
      return null;
    }

    return (
      <div {...rest} ref={forwardRef} className={classes}>
        {error}
      </div>
    );
  }
}
