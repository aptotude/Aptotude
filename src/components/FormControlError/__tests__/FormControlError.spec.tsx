import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoFormControlError } from '../AptoFormControlError';

afterEach(() => {
  cleanup();
});

describe('AptoFormControlError', () => {
  it('Renders null with no data', () => {
    const { container } = render(<AptoFormControlError />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Component with message', () => {
    const { container } = render(<AptoFormControlError error="no bueno" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Component with extra class', () => {
    const { container } = render(
      <AptoFormControlError className="foo" error="no bueno" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
