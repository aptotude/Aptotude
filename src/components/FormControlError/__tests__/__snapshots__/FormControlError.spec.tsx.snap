// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoFormControlError Renders Component with extra class 1`] = `
<div
  class="AptoFormControlError foo"
>
  no bueno
</div>
`;

exports[`AptoFormControlError Renders Component with message 1`] = `
<div
  class="AptoFormControlError"
>
  no bueno
</div>
`;

exports[`AptoFormControlError Renders null with no data 1`] = `null`;
